core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceRequestLogNameTest.java

```diff
@@ -82,4 +75,56 @@ void logNameShouldBeSetOnEarlyFailure() throws Exception {
         assertThat(log.name()).isEqualTo("bar");
         assertThat(log.responseHeaders().status()).isEqualTo(HttpStatus.INTERNAL_SERVER_ERROR);
     }
+
+    @Test
+    void defaultServiceName() throws Exception {
+        WebClient.of(server.httpUri()).get("/ok").aggregate().join();
+
+        final RequestLog log = logs.take().whenComplete().join();
+        assertThat(log.serviceName()).isEqualTo(FooService.class.getName());
+    }
+
+    @Test
+    void customServiceNameWithClass() throws Exception {
+        WebClient.of(server.httpUri()).get("/serviceName/foo").aggregate().join();
+
+        final RequestLog log = logs.take().whenComplete().join();
+        assertThat(log.serviceName()).isEqualTo("MyBarService");
+    }
+
+    @Test
+    void customServiceNameWithMethod() throws Exception {
+        WebClient.of(server.httpUri()).get("/serviceName/bar").aggregate().join();
+
+        final RequestLog log = logs.take().whenComplete().join();
+        assertThat(log.serviceName()).isEqualTo("SecuredBarService");
+    }
+
+    private static class FooService {
+        @Get("/ok")
+        public String foo() {
+            return "OK";
+        }
+
+        @Get("/fail_early")
+        public String bar() {
+            return "Not OK";
+        }
+    }
+
+    @ServiceName("MyBarService")
+    private static class BarService {
+        @Get("/foo")
+        public String foo() {
+            return "OK";
+        }
+
+        @ServiceName("SecuredBarService")
+        @Get("/bar")
+        public String secured(ServiceRequestContext ctx) {
+            ctx.log();
```

Comment:
- ?
- Could you check if the above example (`assert ctx.log().partial().serviceName() == "my-service";`) really works here? If not, please update them to use `whenComplete().thenApply()` which is the recommended way.

Reply:
I think so because the names are set in
https://github.com/line/armeria/blob/173de3301b3694a3153e7c0ded28bb792c4fdc2a/core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java#L246
The `secured(ServiceRequestContext ctx)` is called after this.
https://github.com/line/armeria/blob/173de3301b3694a3153e7c0ded28bb792c4fdc2a/core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java#L258-L263

Reply:
By the way, I agreed that it is not a recommended way to use `RequestLog` because users might copy this code and used it in a decorator.
Let me add an example for a decorator.